### PR TITLE
(NOBIDS) move IndexOldEth1Blocks

### PR DESCRIPTION
**Pull request by MauserBitfly**
### <samp>🤖 Generated by Copilot at 8f2dd48</samp>

Added a new command `index-old-eth1-blocks` to index historical eth1 data into Bigtable. Removed the unused file `cmd/misc/index-old-eth1-blocks/main.go`.

### CODE REVIEW

File: cmd/misc/main.go [hunk 1]
```diff
@@ -245,3 +258,91 @@ func ClearBigtable(family string, key string, dryRun bool, bt *db.Bigtable) {
 		logrus.Infof("%v keys have been deleted", len(deletedKeys))
 	}
 }
+
+func IndexOldEth1Blocks(startBlock uint64, endBlock uint64, batchSize uint64, concurrency uint64, transformerFlag string, bt *db.Bigtable) {
```
Comment (D13ce) [inline cmd/misc/main.go:262]: This should include some sanity checks:
1) `startBlock` < `endBlock`
2) `batchSize` > 0
3) `concurrency` > 0
4) `bt` != nil

These checks should be done first i.e. even before `client, err := rpc.NewErigonClient(utils.Config.Eth1ErigonEndpoint)` is executed.This applies to https://github.com/gobitfly/eth2-beaconchain-explorer/blob/8f2dd48904c8628f64813ced7a612f5bbad8c0b7/cmd/misc/main.go#L270-L275 as well.

File: cmd/misc/index-old-eth1-blocks/main.go [hunk 2]
```diff
@@ -1,207 +0,0 @@
...
-	utils.Config = cfg
-	logrus.WithField("config", *configPath).WithField("version", version.Version).WithField("chainName", utils.Config.Chain.Config.ConfigName).Printf("starting")
-
-	// enable pprof endpoint if requested
-	if utils.Config.Pprof.Enabled {
-		go func() {
-			logrus.Infof("starting pprof http server on port %s", utils.Config.Pprof.Port)
-			logrus.Info(http.ListenAndServe(fmt.Sprintf("localhost:%s", utils.Config.Pprof.Port), nil))
-		}()
-	}
```
Comment (D13ce) [inline cmd/misc/index-old-eth1-blocks/main.go:70]: This feature now is not available anymore. Is that intended?

**Review by D13ce — CHANGES_REQUESTED** [review]

File: cmd/misc/index-old-eth1-blocks/main.go [hunk 3]
```diff
@@ -1,207 +0,0 @@
...
-		if importENSChanges {
-			if err = bt.ImportEnsUpdates(client.GetNativeClient()); err != nil {
-				utils.LogError(err, "error importing ens from events", 0)
-				return
-			}
-		}
-
-		logrus.Infof("indexing of block %v completed", *block)
-		return
-	}
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply cmd/misc/index-old-eth1-blocks/main.go:167]: yes was not necessary as you can achieve the same thing with start-end

File: cmd/misc/index-old-eth1-blocks/main.go [hunk 4]
```diff
@@ -1,207 +0,0 @@
...
-	versionFlag := flag.Bool("version", false, "Print version and exit")
-
-	configPath := flag.String("config", "", "Path to the config file, if empty string defaults will be used")
-
-	flag.Parse()
-
-	if *versionFlag {
-		fmt.Println(version.Version)
-		return
-	}
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply cmd/misc/index-old-eth1-blocks/main.go:47]: don't think its needed

File: cmd/misc/main.go [hunk 5]
```diff
@@ -45,6 +53,11 @@ func main() {
 	flag.Int64Var(&opts.TargetVersion, "target-version", -2, "Db migration target version, use -2 to apply up to the latest version, -1 to apply only the next version or the specific versions")
 	flag.StringVar(&opts.Family, "family", "", "big table family")
 	flag.StringVar(&opts.Key, "key", "", "big table key")
+	flag.Uint64Var(&opts.StartBlock, "blocks.start", 0, "Block to start indexing")
+	flag.Uint64Var(&opts.EndBlock, "blocks.end", 0, "Block to finish indexing")
+	flag.Uint64Var(&opts.DataConcurrency, "data.concurrency", 30, "Concurrency to use when indexing data from bigtable")
+	flag.Uint64Var(&opts.BatchSize, "data.batchSize", 1000, "Batch size")
+	flag.StringVar(&opts.Transformers, "transformers", "", "Komma separated list of transformers used by the eth1 indexer")
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply cmd/misc/main.go:60]: But what if we need it for another action here, then renaming would be tedious. And it's not so far fetched that we use concurrency in other actions as well in the future. 

But I get where you are coming from.

File: cmd/misc/index-old-eth1-blocks/main.go (hunk 2 above)
Comment (MauserBitfly) [reply cmd/misc/index-old-eth1-blocks/main.go:70]: yes - not needed for usecase

File: cmd/misc/main.go (hunk 1 above)
Comment (MauserBitfly) [reply cmd/misc/main.go:262]: added, except for batchSize that defaults maxes 1 - so it's optional

File: cmd/misc/main.go [hunk 6]
```diff
@@ -135,7 +147,8 @@ func main() {
 		CompareRewards(opts.StartDay, opts.EndDay, opts.Validator, bt)
 	case "clear-bigtable":
 		ClearBigtable(opts.Family, opts.Key, opts.DryRun, bt)
-
+	case "index-old-eth1-blocks":
+		IndexOldEth1Blocks(opts.StartBlock, opts.EndEpoch, opts.BatchSize, opts.DataConcurrency, opts.Transformers, bt)
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply cmd/misc/main.go]: good catch ;-)

File: cmd/misc/main.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply cmd/misc/main.go]: Corrected

File: cmd/misc/main.go [hunk 7]
```diff
@@ -260,6 +260,19 @@ func ClearBigtable(family string, key string, dryRun bool, bt *db.Bigtable) {
 }
 
 func IndexOldEth1Blocks(startBlock uint64, endBlock uint64, batchSize uint64, concurrency uint64, transformerFlag string, bt *db.Bigtable) {
+	if endBlock > 0 && endBlock < startBlock {
+		utils.LogError(nil, fmt.Sprintf("endBlock [%v] < startBlock [%v]", endBlock, startBlock), 0)
+		return
+	}
+	if concurrency == 0 {
+		utils.LogError(nil, "concurrency must be greater then 0", 0)
```
Comment (D13ce) [inline cmd/misc/main.go:268]: **Nitpick:**
```suggestion
		utils.LogError(nil, "concurrency must be greater than 0", 0)
```

File: cmd/misc/main.go [hunk 8]
```diff
@@ -260,6 +260,19 @@ func ClearBigtable(family string, key string, dryRun bool, bt *db.Bigtable) {
 }
 
 func IndexOldEth1Blocks(startBlock uint64, endBlock uint64, batchSize uint64, concurrency uint64, transformerFlag string, bt *db.Bigtable) {
+	if endBlock > 0 && endBlock < startBlock {
```
Comment (D13ce) [inline cmd/misc/main.go:263]: I don't understand `endBlock > 0` here. If `endBlock` is 0 and `startBlock` is 1 then this check does not trigger an error even though it should.
Shouldn't it be possible to simply remove the `endBlock > 0`? Why is it needed?

File: cmd/misc/main.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (D13ce) [reply cmd/misc/main.go]: Thanks for your last sentence :stuck_out_tongue_closed_eyes:  :heart:

File: db/ens.go [hunk 9]
```diff
@@ -165,12 +165,18 @@ func (bigtable *Bigtable) TransformEnsNameRegistered(blk *types.Eth1Block, cache
 
 			nameRegistered, err := filterer.ParseNameRegistered(nameLog)
 			if err != nil {
-				utils.LogError(err, fmt.Sprintf("indexing of register event failed parse register event at index [%v] on block [%v]", foundNameIndex, blk.Number), 0)
+				utils.LogError(err, fmt.Sprintf("indexing of register event failed parse register event at tx [%v] index [%v] on block [%v]", i, foundNameIndex, blk.Number), 0)
 				continue
 			}
+
+			if err = verifyName(nameRegistered.Name); err != nil {
+				logger.Warnf("indexing of register event failed because of invalid name at tx [%v] index [%v] on block [%v]: %v", i, foundNameIndex, blk.Number, err)
```
Comment (D13ce) [inline db/ens.go:173]: **Suggestion:** I would add the name to this log entry.

File: db/ens.go [hunk 10]
```diff
@@ -286,6 +297,15 @@ func (bigtable *Bigtable) TransformEnsNameRegistered(blk *types.Eth1Block, cache
 	return bulkData, bulkMetadataUpdates, nil
 }
 
+// Ens names are only supported to a length of 40
+// Source: https://github.com/wealdtech/go-ens/blob/5b323a4ef0472f06c515723b060b166843b9db08/resolver.go#L190
+func verifyName(name string) error {
+	if (strings.HasPrefix(name, "0x") && len(name) > 42) || (!strings.HasPrefix(name, "0x") && len(name) > 40) {
+		return fmt.Errorf("address too long")
```
Comment (D13ce) [inline db/ens.go:304]: **Nitpick:**
```suggestion
		return fmt.Errorf("name too long")
```

**Review by D13ce — COMMENTED** [review]
I have added a question and a few nitpicky comments. ptal

File: db/ens.go (hunk 9 above)
Comment (MauserBitfly) [reply db/ens.go:173]: As these can get really long, I don't want to spam the log file. You can find the event with the other logged informations.

File: cmd/misc/main.go (hunk 8 above)
Comment (MauserBitfly) [reply cmd/misc/main.go:263]: because if endBlock is 0 then we we run from the startBlock to the current one.

**Comment by MauserBitfly** [comment]
> I have added a question and a few nitpicky comments. ptal

thanks updated/commented accordingly

**Review by D13ce — APPROVED** [review]
lgtm

**Pull request merged**